include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,195 @@
+
+namespace alpaka
+{
+    template<typename Pointer, typename Value, typename Idx, typename Dim>
+    struct Accessor;
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<1>>
+    {
+        auto& operator[](Vec<DimInt<1>, Idx> i) const
+        {
+            return (*this)(i[0]);
+        }
+
+        auto& operator[](Idx i) const
+        {
+            return (*this)(i);
+        }
+
+        auto& operator()(Idx i) const
+        {
+            return p[i];
+        }
+
+        Pointer p;
+        Vec<DimInt<1>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<2>>
+    {
+        auto& operator[](Vec<DimInt<2>, Idx> i) const
+        {
+            return (*this)(i[0], i[1]);
+        }
+
+        auto& operator()(Idx y, Idx x) const
+        {
+            return *(reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + y * rowPitchInBytes) + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Vec<DimInt<2>, Idx> extents;
+    };
+
+    template<typename Pointer, typename Value, typename Idx>
+    struct Accessor<Pointer, Value, Idx, DimInt<3>>
+    {
+        auto& operator[](Vec<DimInt<3>, Idx> i) const
+        {
+            return (*this)(i[0], i[1], i[2]);
+        }
+
+        auto& operator()(Idx z, Idx y, Idx x) const
+        {
+            return *(
+                reinterpret_cast<Pointer>(reinterpret_cast<char*>(p) + z * slicePitchInBytes + y * rowPitchInBytes)
+                + x);
+        }
+
+        Pointer p;
+        Idx rowPitchInBytes;
+        Idx slicePitchInBytes;
+        Vec<DimInt<3>, Idx> extents;
+    };
+
+#if 0
+    using Image = cudaTextureObject_t;
+
+    template<typename Value, typename Idx>
+    struct Accessor<Image, Value, Idx, DimInt<1>>
+    {
+        auto operator[](Vec<DimInt<1>, Idx> i) const -> Value
+        {
+            return (*this)(i[0]);
+        }
+
+        auto operator[](Idx i) const -> Value
+        {
+            return (*this)(i);
+        }
+
+        auto operator()(Idx i) const -> Value
+        {
+            return tex1Dfetch(texObj, i);
+        }
+
+        auto operator()(float i) const -> Value
+        {
+            return tex1D(texObj, i);
+        }
+
+        Image texObj;
+        Vec<DimInt<1>, Idx> extents;
+    };
+
+    template<typename Value, typename Idx>
+    struct Accessor<Image, Value, Idx, DimInt<2>>
+    {
+        auto operator[](Vec<DimInt<2>, Idx> i) const -> Value
+        {
+            return (*this)(i[0], i[1]);
+        }
+
+        auto operator()(Idx y, Idx x) const -> Value
+        {
+            return tex1Dfetch(texObj, y * rowPitchInValues + x);
+        }
+
+        auto operator()(float y, float x) const -> Value
+        {
+            return tex2D(texObj, x, y);
+        }
+
+        Image texObj;
+        Idx rowPitchInValues;
```

Comment:
Since we cannot do the pointer arithemtic ourselves for `tex1Dfetch`, we need to specify the pitch in number of objects of type `Value` instead of bytes.  Since the types for `Value` are restricted by e.g. CUDA and CUDA also has control of the pitch used in `cudaMallocPitch`, CUDA can ensure it works. I do not know if it does however, and we should assert when allocating a texture object that the used pitch is a multiple of the texel type's size.